### PR TITLE
Implement basic test coverage

### PR DESCRIPTION
**Pull request by kzantow**

### CODE REVIEW

File: src/github/SyftGithubAction.ts [hunk 1]
```diff
@@ -197,10 +197,10 @@ export async function uploadSbomArtifact(contents: string): Promise<void> {
  */
 function getBooleanInput(name: string, defaultValue: boolean): boolean {
   const val = core.getInput(name);
-  if (val === "") {
+  if (val === undefined || val === "") {
```
Comment (luhring) [inline src/github/SyftGithubAction.ts:200]: Is `null` a possible value?

File: tests/mocks.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
+export const artifacts: any = [];
+
+export const assets: any = [];
+
+export const workflowRun: WorkflowRun = {
+  id: 4309583450,
+} as any;
+
+export let inputs: { [key: string]: string } = {};
```
Comment (luhring) [inline tests/mocks.ts:9]: This looks like we're maintaining and exposing some "global state" that's later set and read by individual tests. Is that true?

If it is, I'm wondering if this limits the ability for tests to run in parallel, and also, global state can sometimes lead to code being trickier to debug and harder to follow.

What are your thoughts?

File: src/github/GithubClient.ts [hunk 3]
```diff
@@ -40,11 +40,65 @@ export interface WorkflowRun {
...
+ * Suppress info output by redirecting to debug
+ * @param fn function to call for duration of output suppression
+ */
+async function suppressOutput<T>(fn: () => Promise<T>): Promise<T> {
+  const info = core.info;
+  try {
+    try {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      core.info = core.debug;
```
[the remark this answers is not in the document]
Comment (kzantow) [reply src/github/GithubClient.ts:53]: If something calls `Object.freeze`, properties can no longer be set. Apparently something along these lines is happening with the jest mocks in the tests.

File: .eslintrc.json [hunk 4]
```diff
@@ -1,65 +1,21 @@
...
-      }]
-    },
-    "env": {
-      "node": true,
-      "commonjs": true,
-      "es6": true,
-      "jest/globals": true
-    }
+  "plugins": ["import"],
+  "extends": ["airbnb-typescript", "prettier"],
```
[the remark this answers is not in the document]
Comment (kzantow) [reply .eslintrc.json:3]: Roughly, yes. The problem is... when I merged some of the dependabot PRs, these lint rules go all out of whack. It's _still_ not in a great state. I need to revisit this but don't want to hold up this PR for it. Added a ticket for it here: https://github.com/anchore/sbom-action/issues/48

File: src/github/SyftGithubAction.ts (hunk 1 above)
Comment (kzantow) [reply src/github/SyftGithubAction.ts:200]: `null` is not really ever going to be set here. As a Github action, it's always returning empty string for unset properties. I was just lazy with tests and left properties undefined in the mocks. I suppose I could instead return `""`. This just seemed like a logical check to add.

File: src/github/GithubClient.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (luhring) [reply src/github/GithubClient.ts:53]: Ah, interesting! TIL

File: tests/mocks.ts (hunk 2 above)
Comment (kzantow) [reply tests/mocks.ts:9]: This could be a problem, let me see if I can relatively easily fix this.

Comment (kzantow) [reply tests/mocks.ts]: Ok, moved the state so it is no longer global, only per `getMocks` invocation. Each module is effectively executed in sequence in jest, so the module-level scope should not cause problems.

**Review by luhring — APPROVED** [review]

**Pull request merged**